### PR TITLE
support ATM forecast only on Azure

### LINKED ISSUE #2826: Add support for forecast-only runs on Azure
### What new functionality do you need?

Add support for forecast-only runs on Azure

### What are the requirements for the new functionality?

Add/Modify files to support for forecast-only runs on Azure.

### Acceptance Criteria

All current working cases continue work, and add new capability to run on forecast only on Azure.

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by weihuang-jedi**
This PR will support ATM forecast only run on Azure, and solve issue: #2826.

# Description

Support ATM forecast only runs on Azure. This work is based on the work of support ATM forecast only run on AWS.

  Resolves #2826
  Refs #2711 

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? YES
- Does this change require an update to any of the following submodules? NO (If YES, please add a link to any PRs that are pending.)
  - [ ] EMC verif-global
  - [ ] GDAS
  - [ ] GFS-utils
  - [ ] GSI
  - [ ] GSI-monitor
  - [ ] GSI-utils
  - [ ] UFS-utils
  - [ ] UFS-weather-model
  - [ ] wxflow

# How has this been tested?

- Clone and build on Azure cluster
- Forecast-only on Azure

# Checklist
- [x ] Any dependent changes have been merged and published
- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] My changes generate no new warnings
- [ x] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: env/AZUREPW.env [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    exit 2
+fi
+
+if [[ "${step}" = "fcst" ]] || [[ "${step}" = "efcs" ]]; then
+
+    export I_MPI_FABRICS=shm:ofi
+    export I_MPI_OFI_PROVIDER=tcp
+   #export I_MPI_DEBUG=6
+
+    export launcher="srun --mpi=pmi2 -l"
```
Comment (aerorahul) [inline env/AZUREPW.env:38]: There is no need to export this.

File: env/AZUREPW.env [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+elif [[ "${step}" = "epos" ]]; then
+
+    export NTHREADS_EPOS=${NTHREADSmax}
+    export APRUN_EPOS="${APRUN}"
+
+elif [[ "${step}" = "fit2obs" ]]; then
+
+    export NTHREADS_FIT2OBS=${NTHREADS1}
+    export MPIRUN="${APRUN}"
```
Comment (aerorahul) [inline env/AZUREPW.env]: These are not forecast jobs.  Please remove.

File: parm/config/gfs/config.base [hunk 3]
```diff
@@ -483,9 +483,9 @@ export OFFSET_START_HOUR=0
 # Number of regional collectives to create soundings for
 export NUM_SND_COLLECTIVES=${NUM_SND_COLLECTIVES:-9}
 
-# The tracker, genesis, and METplus jobs are not supported on AWS yet
-# TODO: we should place these in workflow/hosts/awspw.yaml as part of AWS setup, not for general.
-if [[ "${machine}" == "AWSPW" ]]; then
+# The tracker, genesis, and METplus jobs are not supported on CSPs yet
+# TODO: we should place these in workflow/hosts/awspw.yaml as part of AWS/AZURE setup, not for general.
+if [[ "${machine}" == "AWSPW" || "${machine}" == "AZUREPW" ]]; then
```
Comment (aerorahul) [inline parm/config/gfs/config.base]:
```suggestion
if [[ "${machine}" =~ "PW" ]]; then
```
Would this work?
Or:
```suggestion
if [[ "${machine}" =~ ^(AWSPW|AZUREPW)$ ]]; then
```

**Review by aerorahul — COMMENTED** [review]
very minor comments. most look good.

File: env/AZUREPW.env (hunk 2 above)
Comment (weihuang-jedi) [reply env/AZUREPW.env]: removed.

File: parm/config/gfs/config.base (hunk 3 above)
Comment (weihuang-jedi) [reply parm/config/gfs/config.base]: Use the first suggestion.

File: env/AZUREPW.env (hunk 1 above)
Comment (weihuang-jedi) [reply env/AZUREPW.env:38]: Doing more check, before removing.

Comment (weihuang-jedi) [reply env/AZUREPW.env:38]: Checked. Remove export I_MPI_xx env. var.

**Review by aerorahul — APPROVED** [review]
looks good.
Thanks @weihuang-jedi

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**